src/actions/player/playerTrendsActions.js

```diff
@@ -33,33 +36,60 @@ export const getPlayerTrendsError = (payload, id) => ({
   id,
 });
 
+const getCumalativeSums = (matches, fieldName) =>
+  matches.reduce((cumulativeList, match, index) => {
+    if (cumulativeList.length > 0) {
+      const prevTotal = cumulativeList[index - 1];
+      cumulativeList.push(prevTotal + match[fieldName]);
+    } else {
+      cumulativeList.push(match[fieldName]);
+    }
+    return cumulativeList;
+  }, []);
+
+const getWinRateCumlativeSums = matches =>
+  matches.reduce((cumulativeList, match, index) => {
+    // 1 for win 0 for loss
+    const winInt = match.radiant_win === isRadiant(match.player_slot) ? 1 : 0;
+    if (index === 0) {
+      cumulativeList.push(winInt);
+    } else {
+      const prevRate = cumulativeList[index - 1];
+      const newRate = ((prevRate * index) + winInt) / (index + 1);
+      cumulativeList.push(newRate);
+    }
+
+    return cumulativeList;
+  }, []);
+
 export const getPlayerTrends = (playerId, options = {}, fieldName) => (dispatch) => {
   dispatch(getPlayerTrendsRequest(playerId));
   return fetch(`${API_HOST}${getUrl(playerId, options, url)}`)
     .then(response => response.json())
     .then((response) => {
```

Comment:
can you just .map() the response to add a field called win_rate with a value of 0 or 1?  Then you can just reuse the existing logic?